swupd/delta.go

```diff
@@ -0,0 +1,164 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest *Manifest, statedir string, from, to uint32) ([]*File, error) {
+	var c config
+	c, err := getConfig(statedir)
```

Comment:
Seems we will be reading the config a LOT of times.

Can we just pass the config in? Probably help with unit testing as well.

Reply:
Oops, wrong place.

Reply:
That might make sense, could pass it in to `CreateManifests` as well.

Reply:
If this is used for Manifest.Full it will only be read once, otherwise yes..it would be called a lot. I can change this to pass in regardless though.